packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+import { useAppContext } from '../../context/app';
+import { useUserContext } from '../../context/user';
+import { useCartContext } from '../../context/cart';
+
+export const useCheckoutPage = props => {
+    const { createCartMutation, getCartDetailsQuery } = props;
+
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ isSignedIn }] = useUserContext();
+    const [
+        { isEmpty },
+        { createCart, getCartDetails, removeCart }
+    ] = useCartContext();
+
+    const [fetchCartId] = useMutation(createCartMutation);
+    const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
+
+    const handleSignIn = useCallback(() => {
+        toggleDrawer('nav');
+    }, [toggleDrawer]);
+
+    /**
+     * TODO. This needs to change to checkout mutations
+     * or other mutations once we start checkout development.
+     *
+     * For now we will be using removeCart and createCart to
+     * simulate a cart clear on Place Order button click.
+     */
+    const cleanUpCart = useCallback(async () => {
+        await removeCart();
+
+        await createCart({
+            fetchCartId
+        });
+
+        await getCartDetails({ fetchCartId, fetchCartDetails });
+    }, [removeCart, createCart, getCartDetails, fetchCartId, fetchCartDetails]);
+
+    /**
+     * Using local state to maintain these booleans. Can be
+     * moved to checkout context in the future if needed.
+     *
+     * These are needed to track progree of checkout steps.
+     */
+    const [shippingInformationDone, updateShippingInformationDone] = useState(
+        false
+    );
+    const [shippingMethodDone, updateShippingMethodDone] = useState(false);
+    const [paymentInformationDone, updatePaymentInformationDone] = useState(
+        false
+    );
+    const [orderPlaced, updateOrderPlaced] = useState(false);
+
+    const setShippingInformationDone = useCallback(
+        () => updateShippingInformationDone(true),
+        [updateShippingInformationDone]
+    );
+    const setShippingMethodDone = useCallback(
+        () => updateShippingMethodDone(true),
+        [updateShippingMethodDone]
+    );
+    const setPaymentInformationDone = useCallback(
+        () => updatePaymentInformationDone(true),
+        [updatePaymentInformationDone]
+    );
+    /**
+     * TODO
+     */
+    const placeOrder = useCallback(async () => {
+        await cleanUpCart();
+        updateOrderPlaced(true);
+    }, [cleanUpCart, updateOrderPlaced]);
+
+    return [
+        {
+            isGuestCheckout: !isSignedIn,
+            isCartEmpty: isEmpty,
```

Comment:
If we change to use a checkout page graphql query here we can check for `total_quantity` like we do on the cart page. If you could make that change here it'll be one less thing we have to refactor later.

See https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/CartPage/useCartPage.js#L37

Reply:
> I'm not sure there's a follow-up ticket for checkout refactor but I definitely don't want to scope creep you in this ticket. I'll make a ticket and reference this comment.
> 
> Made this: https://jira.corp.magento.com/browse/PWA-395

Let's handle such tasks in the ticket you created last week.